ci/before_install.sh

```diff
@@ -10,3 +10,13 @@ wget "https://bouncycastle.org/download/${BCPROV_FILENAME}"
 sudo mv $BCPROV_FILENAME /usr/lib/jvm/java-7-openjdk-amd64/jre/lib/ext
 sudo perl -pi.bak -e 's/^(security\.provider\.)([0-9]+)/$1.($2+1)/ge' /etc/java-7-openjdk/security/java.security
 echo "security.provider.1=org.bouncycastle.jce.provider.BouncyCastleProvider" | sudo tee -a /etc/java-7-openjdk/security/java.security
+
+# Install Thrift 0.9.3
+sudo apt-get update -qq
```

Comment:
Why do you need to install `thrift` binaries? if only for the codegen, might be better to include the generated code in the diff.

Reply:
Two options here:
1. Yes, installing thrift binaries here are for codegen, but since we compile thrift files as part of the project build, we are also allowing Travis to test our build process as a developer would build the project. However, installing thrift binaries are by far the most time consuming part of a Travis build.
1. The other option is that we just check in the generated code, which means we'd have to specify Travis-specific build parameters to not have it try to compile thrift.

Thoughts?

Reply:
I don't think developers are going to be regening thrift code often at all, so (1) seems excessive. I suggest (2).

Reply:
I've gone ahead and made Thrift code generation skipped by default in addition to checking in the generated source code. Code generation can be triggered by passing a property (documented in [DEVELOPERS.md](DEVELOPERS.md)).

Reply:
Solid